src/libp2p_swarm_sup.erl

```diff
@@ -87,7 +87,7 @@ init([Name, Opts]) ->
     ets:insert(TID, {?ADDRESS, libp2p_crypto:pubkey_to_bin(PubKey)}),
     GroupDeletePred = libp2p_config:get_opt(Opts, group_delete_predicate, fun(_) -> false end),
 
-    SupFlags = {one_for_all, 3, 10},
+    SupFlags = {one_for_one, 3, 10},
```

Comment:
@madninja @macpie can you think of any reason why we shouldn't make this change?  peerbook failures shouldn't really take the rest of the swarm components down, from what I can tell, but I don't know if this was done for a particular reason.

Reply:
The peerbook will be it's own thing shortly once @andymck is done with his refactor, too.

Reply:
My main concern would be that the children of this sup aren't ready to be restarted without the whole supervisor coming back up to give them all the handles/state they need.

Reply:
yeah i am changing this area.  the peerbook is in its own tree, and the swarm components in their own tree.  Both trees reside under a top level swarm specific 'super sup'.  Still a few issues to iron out atm with handles but I am currently using named tables ( so the ETS TID is now just the swarm name ) and thus any process can get the peerbook handle as long as they know the swarm name...

Reply:
I am also considering using registered names for core processes,  based around the swarm name,  to avoid having to know the PID ( for example for the peerbook )